samples/java-spring-cinema-booking-choreography/README.md

```diff
@@ -0,0 +1,1379 @@
+# Cinema Booking sample
```

Comment:
There are a lot of headings and nested headings. I suggest adding a table of contents to make it easier to understand the structure of the content and move from section to section.

```suggestion
# Cinema Booking sample

## Table of Contents
<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
  - [Prerequisites](#prerequisites)
  - [Kalix SDK](#kalix-sdk)
- [Designing Kalix Services](#designing-kalix-services)
  - [Use case definition](#use-case-definition)
  - [Event storming](#event-storming)
    - [Entities / Aggregate Roots](#entities--aggregate-roots)
    - [Commands & Domain Events](#commands--domain-events)
  - [Bounded contexts](#bounded-contexts)
    - [Show](#show)
    - [Wallet](#wallet)
  - [Context map](#context-map)
  - [Non-functional requirements](#non-functional-requirements)
  - [Kalix components](#kalix-components)
    - [Show](#show-1)
    - [Wallet](#wallet-1)
    - [Context map](#context-map-1)
- [Kickstart a Kalix development project](#kickstart-a-kalix-development-project)
  - [Maven archetype](#maven-archetype)
  - [Import the generated project into your IDE/editor](#import-the-generated-project-into-your-ideeditor)
  - [Adding additional, non-mandatory dependencies](#adding-additional-non-mandatory-dependencies)
  - [Java 21](#java-21)
- [Develop Kalix components for the "Show" bounded context](#develop-kalix-components-for-the-show-bounded-context)
  - [Describe Data structures](#describe-data-structures)
    - [API data structures](#api-data-structures)
    - [Domain Events data structure](#domain-events-data-structure)
    - [Show entity state data structure and business domain logic encapsulation](#show-entity-state-data-structure-and-business-domain-logic-encapsulation)
  - [Develop the "Show" Entity](#develop-the-show-entity)
  - [Show Entity unit test](#show-entity-unit-test)
    - [Run unit test](#run-unit-test)
  - [Run & test locally](#run--test-locally)
    - [Run](#run)
    - [Test](#test)
  - [Developing the "Shows By Availability" View](#developing-the-shows-by-availability-view)
  - [View integration test](#view-integration-test)
    - [Run integration test](#run-integration-test)
  - [Developing the "Reservation" Entity](#developing-the-reservation-entity)
  - [Developing the `FoldShowEventsToReservation` Kalix Action](#developing-the-foldshoweventstoreservation-kalix-action)
  - [Entity and Action interaction integration test](#entity-and-action-interaction-integration-test)
    - [Run integration test](#run-integration-test-1)
- [Developing Kalix components for "Wallet" bounded context](#developing-kalix-components-for-wallet-bounded-context)
  - [Describing Data structures](#describing-data-structures)
    - [API data structures](#api-data-structures-1)
    - [Domain Events data structure](#domain-events-data-structure-1)
    - [Wallet entity state data structure and business domain logic encapsulation](#wallet-entity-state-data-structure-and-business-domain-logic-encapsulation)
  - [Developing the "Wallet" Entity](#developing-the-wallet-entity)
  - [Wallet Entity unit test](#wallet-entity-unit-test)
    - [Run unit test](#run-unit-test-1)
  - [Developing "Show - Wallet asynchronous event driven communication" (eventing)](#developing-show---wallet-asynchronous-event-driven-communication-eventing)
    - [`ChargeForReservation` Action](#chargeforreservation-action)
    - [Complete Reservation Action](#complete-reservation-action)
  - [End to end integration test](#end-to-end-integration-test)
    - [Run integration test](#run-integration-test-2)
- [Deploying](#deploying)
  - [Test service in production](#test-service-in-production)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->
```